### PR TITLE
Bah 781 individuals involved

### PR DESCRIPTION
**Pull request by aaronjlech**
## Description
This adds the individuals involved yes/no page and the followUp page to add individuals involved within an incident.

## Testing done
- [x] Unit testing
- [x] Local testing

## Screenshots
image
image

## Acceptance criteria
- [x] User is able to populate First Name, Middle Name, Last Name
- [x] User is able to populate individual Description
- [x] User is able to indicate if individual is a Service Member and what rank, unit, date of injury
- [x] User is able to indicate how the other individual was injured
- [x] If 'Other' is selected then user is able to add free text if desired
- [x] User is able to add another individual and repeat the entry process for another individual until all individuals are entered

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by imeecuison — APPROVED** [review]
Looks good!

File: src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export default function IndividualsInvolvedCard({ formData }) {
+  const { name } = formData;
+  const serviceMemberTitle = formData['view:serviceMember']
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx]: I find these ternaries kind of hard to read. What about:

```js
let displayTitle;

if (name.first || name.last) {
  displayTitle = `${name.first || ''} ${name.last || ''}`
} else {
  displayTitle = formData['view:serviceMember'] 
    ? 'Servicemember'
    : 'Civilian';
}
```

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    [`incident${index}`]: {
+      type: 'object',
+      properties: {
+        personInvolved: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              name: {
+                $ref: '#/definitions/fullName',
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Is this actually a definition in the schema right now?

File: src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx (hunk 1 above)
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx]: Sure I'm all for readability. The Linter with line length does effect lengthy turneries!

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js (hunk 2 above)
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Yeah.

File: src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export default function IndividualsInvolvedCard({ formData }) {
+  const { name } = formData;
+  let displayTitle;
+  if (name.first || name.last) {
+    displayTitle = `${name.first || ''} ${name.last || ''}`;
+  } else {
+    displayTitle = formData['view:serviceMember']
+      ? 'Servicemember'
+      : 'Civilian';
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/components/IndividualsInvolvedCard.jsx]: That's the idea. `formData['view:serviceMember']` will be undefined and Civilian will be set as the display name.

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 4]
```diff
@@ -61,9 +67,22 @@ export function createFormConfig781(iterations) {
...
+      [`individualsInvolvedFollowUp${index}`]: {
+        title: `${numberToWords[index]} PTSD individuals Involved`,
+        path: `disabilities/ptsd-individuals-involved-questions-${index}`,
+        depends: isAddingIndividuals(index),
+        uiSchema: individualsInvolvedFollowUp.uiSchema(index),
+        schema: individualsInvolvedFollowUp.schema(index),
+      },
       [`ptsdAdditionalEvents${index}`]: {
-        title: `${numberToWords[index]} Combat PTSD Additional events.`,
+        title: `${numberToWords[index]} PTSD additional events.`,
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: We're going with `assault` as the identifier betweeen 781(no identifier) and 781a(assault).

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 5]
```diff
@@ -61,9 +67,22 @@ export function createFormConfig781(iterations) {
         uiSchema: incidentUnitAssignment.uiSchema(index),
         schema: incidentUnitAssignment.schema(index),
       },
-      // This should be the last page in the config loop
+      [`individualsInvolved${index}`]: {
+        title: `${numberToWords[index]} PTSD individuals Involved yes/no`,
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: Good point, this page actually needs to be set to `hideOnReview: true`

File: src/applications/disability-benefits/all-claims/config/schema.js [hunk 6]
```diff
@@ -520,6 +520,20 @@ const schema = {
       },
       required: ['from', 'to'],
     },
+    fullName: {
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/config/schema.js]: Referenced this from another form schema... 🤷‍♂️ 

from `src/applications/burials/config`

```
  defaultDefinitions: {
    fullName,
    centralMailVaFile,
    ssn,
    date,
    usaPhone,
    dateRange,
  },
```

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  [`incident${index}`]: {
+    personInvolved: {
+      'ui:title': ' ',
+      'ui:options': {
+        itemName: 'Individual',
+        viewField: IndividualsInvolvedCard,
+      },
+      items: {
+        'ui:options': {
+          updateSchema: (formData, schema, itemsUISchema, itemIndex) => {
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Yeah!

File: src/applications/disability-benefits/all-claims/content/individualsInvolved.jsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    Or
+    <br />
+    <br />
+    Description of the person
+  </p>
+);
+
+export const individualsDescription = (
+  <div>
+    {individualsInvolved}
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/content/individualsInvolved.jsx]: Used in another file.

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+                  itemIndex,
+                  'injuryDeath',
+                ],
+                formData,
+                '',
+              ) === 'Other';
+
+            if (serviceMemberSelected) {
+              schemaProps = {
+                ...schemaProps,
```
[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: I see what you're saying...but somehow I don't get this side effect of it showing up after no is selected...updateSchema magic?

[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Resolved with `'ui:collapsed: true'` 😄

File: src/applications/disability-benefits/all-claims/config/781/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: Isn't this the title that gets displayed on the page to the veteran? I'm not sure that adding `yes/no` to the title is consistent with what we've done elsewhere.

If this page is marked as `hideOnReview`, then will the user be able to change their answer on the review page?

Also, I just noticed `Involved` is capitalized. Probably shouldn't be, right? :male_detective:

[the remark this answers is not in the document]
Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: Yeah looking at the review page now, I think some things might need to be updated/thought out as far as titles are concerned. Good call though, this should definitely be on the review page.

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+  properties: {
+    [`incident${index}`]: {
+      type: 'object',
+      properties: {
+        personInvolved: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              name: {
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Added this schema hardcoded, the updated `fullName` definition in `vets-json-schema` has suffix(that I don't need) and has first and last required(don't need). We'll be updated our 781 schema inside `vets-json-schema` and then porting those properties into the pages @dmccurdybah is currently working on that.

**Review by cvalarida — APPROVED** [review]
Looks good. 554 additions gets pretty heavy to review, though, so hopefully I didn't miss too much. :grimacing: :crossed_fingers:

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 11]
```diff
@@ -90,16 +90,12 @@ export const schema = index => ({
                 properties: {
                   first: {
                     type: 'string',
-                    minLength: 1,
-                    maxLength: 30,
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Will this cause any problem down the line if a user enters a name that's 3,000 characters long?

Comment (aaronjlech) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js]: Due to the sensitivity of the form, the backend will generate an overflow page for anything that exceeds the limit for the form field (PDF form field).

**Pull request merged**